File: nattybear/level1/문자열-내-마음대로-정렬하기/Solution.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
  public String[] solution(String[] strings, int n) {
    return Arrays
      .stream(strings)
      .sorted(new Comparator<String>() {
        public int compare(String s1, String s2) {
          char c1 = s1.charAt(n);
          char c2 = s2.charAt(n);
          int result = c1 - c2;
          if (result != 0)
            return result;
          else
            return s1.compareTo(s2);
        }
      })
      .toArray(String[]::new);
  }
}

File: nattybear/level1/문자열-내-마음대로-정렬하기/solution.js
function solution(strings, n) {
  strings.sort((s1, s2) => {
    const c1 = s1.charCodeAt(n);
    const c2 = s2.charCodeAt(n);
    const result = c1 - c2;
    if (result != 0)
      return result;
    else
      return s1.localeCompare(s2);
  });
  return strings;
}
